instrumentation/grizzly-2/src/main/java/io/opentelemetry/auto/instrumentation/grizzly/GrizzlyHttpHandlerInstrumentation.java

```diff
@@ -66,39 +64,44 @@ public boolean defaultEnabled() {
   public static class HandleAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static AgentScope methodEnter(@Advice.Argument(0) final Request request) {
+    public static SpanScopePair methodEnter(@Advice.Argument(0) final Request request) {
       if (request.getAttribute(SPAN_ATTRIBUTE) != null) {
         return null;
       }
 
-      final Context parentContext = propagate().extract(request, GETTER);
-      final AgentSpan span = startSpan("grizzly.request", parentContext);
+      final Span.Builder spanBuilder = TRACER.spanBuilder("grizzly.request");
+      try {
+        final SpanContext extractedContext = TRACER.getHttpTextFormat().extract(request, GETTER);
+        spanBuilder.setParent(extractedContext);
+      } catch (final IllegalArgumentException e) {
```

Comment:
hmm. If this is the API that we're providing for this in the opentelemetry-java API, it makes me a little sad. Is there a better way to do this than requiring everyone to catch this?

Reply:
see https://github.com/open-telemetry/opentelemetry-java/issues/767

Reply:
ah! sweet. awesome. I hadn't connected the dots.